### PR TITLE
Python3: fix NVDA's logging infrastructure

### PR DESCRIPTION
**Pull request by michaelDCurran**
### Link to issue number:
None.

### Summary of the issue:
In Python3: several issues cause NVDA's logging to fail:
* NVDA makes use of logging._levelNames which no longer exists in Python3.
* When NvDA calculates a code path when logging, it accesses a function's code object via its func_code property. In Python3 this is now ```__code__```.
* As strings are now unicode, and files that are opened in text mode deal in unicode, NVDA's logHandler.FileHandler no longer works. 
 
### Description of how this pull request fixes the issue:
* No longer use logging._levelNames, rather use logging.getLevelName.
* Changed logHandler.getCodePath to use ```__code__``` instead of func_code.
* logHandler.FileHandler now inherits from logging.Filehandler rather than the lower-level StreamHandler, and no longer converts to utf8 before writing. Rather the File object itself handles the utf8 encoding.

### Testing performed:
Started NVDA under Python3. Info messages and tracebacks were successfully logged.

### Known issues with pull request:
None.

### Change log entry:
None.

Section: New features, Changes, Bug fixes

### CODE REVIEW

File: source/logHandler.py [hunk 1]
```diff
@@ -13,7 +13,6 @@
 from encodings import utf_8
 import logging
 # #7105: Python 3 split the following into two dictionaries.
```
Comment (leonardder) [inline source/logHandler.py]: This comment seems now obsolete.

File: source/logHandler.py [hunk 2]
```diff
@@ -78,7 +77,7 @@ def getCodePath(f):
 				if not member:
 					continue
 				memberType=type(member)
-				if memberType is FunctionType and member.func_code is f.f_code:
```
Comment (leonardder) [inline source/logHandler.py]: I think I prefer something like `if isinstance(member, FunctionType)` while at it. Don't insist, though.

File: source/logHandler.py [hunk 3]
```diff
@@ -78,7 +77,7 @@ def getCodePath(f):
 				if not member:
 					continue
 				memberType=type(member)
-				if memberType is FunctionType and member.func_code is f.f_code:
+				if memberType is FunctionType and member.__code__ is f.f_code:
 					# the function was found as a standard method
 					className=cls.__name__
 				elif memberType is classmethod and type(member.__func__) is FunctionType and member.__func__.func_code is f.f_code:
```
Comment (leonardder) [inline source/logHandler.py]: Also note this line if you touch the former.

File: source/logHandler.py [hunk 4]
```diff
@@ -239,19 +227,10 @@ def handle(self,record):
 				nvwave.playWaveFile("waves\\error.wav")
 			except:
 				pass
-		return logging.StreamHandler.handle(self,record)
+		return logging.FileHandler.handle(self,record)
```
Comment (leonardder) [inline source/logHandler.py]: Can't you just call super here, or do you really want to be explicit?

**Review by LeonarddeR — APPROVED** [review]
Just a few comments

**Review by feerrenrut — CHANGES_REQUESTED** [review]
There are still a few references to `func_code` in this file, could you check the rest of the code-base while you are at it? Could you also check for `logging._levelNames` in the rest of the code-base too?

Note, there are a few review comments from Leonard as well.

**Comment by michaelDCurran** [comment]
I had previously grepped for any other uses of _levelNames and corrected them. There are none left.

**Comment by michaelDCurran** [comment]
I have addressed review comments. Though didn't change  the 'is' to an isinstance as this was existing code and I'm trying to avoid changing anything not necessary to change due to Python 3.

**Review by feerrenrut — APPROVED** [review]

**Pull request merged**